pandas/tests/reshape/test_pivot.py

```diff
@@ -65,6 +66,48 @@ def test_pivot_table(self):
             index + [columns])['D'].agg(np.mean).unstack()
         tm.assert_frame_equal(table, expected)
 
+    def test_pivot_table_categorical_observed_equal(self, observed):
+        # issue #24923
+        df = pd.DataFrame({'col1': list('abcde'),
+                           'col2': list('fghij'),
+                           'col3': [1, 2, 3, 4, 5]})
+
+        df.col1 = df.col1.astype('category')
+        df.col2 = df.col1.astype('category')
+
+        expected = df.pivot_table(index='col1', values='col3',
+                                  columns='col2', aggfunc=np.sum,
+                                  fill_value=0)
+
+        result = df.pivot_table(index='col1', values='col3',
+                                columns='col2', aggfunc=np.sum,
+                                fill_value=0, observed=observed)
+
+        tm.assert_frame_equal(result, expected)
+
+    def test_pivot_table_categorical_observed_speed(self):
+        # issue #24923
+        df = pd.DataFrame({'col1': list('abcde'),
+                           'col2': list('fghij'),
+                           'col3': [1, 2, 3, 4, 5]})
+
+        df.col1 = df.col1.astype('category')
+        df.col2 = df.col1.astype('category')
+
+        start_time_observed_false = time.time()
```

Comment:
why do you have time in here? these types of things are handled by the asv suite. i would remove this entire test.

Reply:
I'm unsure how to otherwise address the test requirements @WillAyd highlighted. The only measurable outcome that observed changes is the time of execution. Have now removed.

Happy to listen to suggestions.

Reply:
you have an asv result, yes? IOW if you timeit under master and then under the PR

Reply:
@benjaminr right I wasn't asking for a test on timing as the ASV will handle that. I was asking for a test where the data makes a difference when this argument is supplied.

Right now the existing test gives the same result whether or not observed is True or False, so it doesn't test that adding this actually does anything for the result explicitly. Can you not come up with a test and data where the keyword would yield different results and test that explicitly?

Reply:
Yes, I added an additional benchmark for this to asv_bench/benchmarks/reshape.py in 5c62063.

I've just added an additional benchmark where observed would default to False, with no kwarg being passed.

**Master**
```
[ 92.59%] ··· reshape.PivotTable.time_pivot_table_categorical                                                                             45.5±2ms
[ 93.52%] ··· reshape.PivotTable.time_pivot_table_categorical_observed                                                                      failed
```
Fails because the kwarg doesn't exist in the master branch yet.

**feature/pivot_table_groupby_observed**
```
[ 67.59%] ··· reshape.PivotTable.time_pivot_table_categorical                                                                             52.9±3ms
[ 68.52%] ··· reshape.PivotTable.time_pivot_table_categorical_observed                                                                    19.1±1ms
```
Speed increase is demonstrated here.